src/libs/actions/Session/index.js

```diff
@@ -23,6 +23,32 @@ Onyx.connect({
     callback: val => credentials = val || {},
 });
 
+/**
+ * Manage push notification subscriptions on sign-in/sign-out.
+ *
+ * We would normally do this when AuthScreens mounts and unmounts. However on Android,
+ * AuthScreens unmounts when the app is closed with the back button so we manage the
+ * push subscription when the session changes here.
```

Comment:
```suggestion
 * On Android, AuthScreens unmounts when the app is closed with the back button so we manage the
 * push subscription when the session changes here.
```

I think this is a good enough warning and feels like we are standing by this solution a bit more. After working through the problem it sounds really undesirable to use `AuthScreens` for this.